chainstate/tx-verifier/src/transaction_verifier/hierarchy.rs

```diff
@@ -98,38 +99,39 @@ impl<C, S: TransactionVerifierStorageRef, U: UtxosView, A: PoSAccountingView>
     fn get_accounting_undo(
         &self,
         id: Id<Block>,
-    ) -> Result<Option<AccountingBlockUndo>, TransactionVerifierStorageError> {
+    ) -> Result<Option<AccountingBlockUndo>, <Self as TransactionVerifierStorageRef>::Error> {
         match self.accounting_block_undo.data().get(&TransactionSource::Chain(id)) {
             Some(v) => Ok(Some(v.undo.clone())),
             None => self.storage.get_accounting_undo(id),
         }
     }
 }
 
-impl<C, S: TransactionVerifierStorageRef, U: UtxosView, A: PoSAccountingView> UtxosStorageRead
+impl<C, S: TransactionVerifierStorageRef, U: UtxosView, A> UtxosStorageRead
     for TransactionVerifier<C, S, U, A>
 {
-    fn get_utxo(&self, outpoint: &OutPoint) -> Result<Option<utxo::Utxo>, storage_result::Error> {
-        Ok(self.utxo_cache.utxo(outpoint))
+    type Error = U::Error;
+
+    fn get_utxo(&self, outpoint: &OutPoint) -> Result<Option<utxo::Utxo>, Self::Error> {
+        self.utxo_cache.utxo(outpoint)
     }
 
-    fn get_best_block_for_utxos(&self) -> Result<Option<Id<GenBlock>>, storage_result::Error> {
+    fn get_best_block_for_utxos(&self) -> Result<Option<Id<GenBlock>>, Self::Error> {
         Ok(Some(self.best_block))
     }
 
-    fn get_undo_data(
-        &self,
-        id: Id<Block>,
-    ) -> Result<Option<UtxosBlockUndo>, storage_result::Error> {
+    fn get_undo_data(&self, id: Id<Block>) -> Result<Option<UtxosBlockUndo>, Self::Error> {
         match self.utxo_block_undo.data().get(&TransactionSource::Chain(id)) {
             Some(v) => Ok(Some(v.undo.clone())),
-            None => self.storage.get_undo_data(id),
+            None => self.storage.get_undo_data(id).map_err(|_| todo!("PR")),
```

Comment:
?

Reply:
Must have overlooked this one, will fix

Reply:
If you add a comment `// TODO(PR)` instead then CI won't pass unless you fix it. If you want to do it with this format that you chose `todo!("PR")`, please add it to `codecheck.py` so that it helps you not to forget.

Reply:
> What's confusing to me is that it has not been caught by `codecheck.py`: [relevant part](https://github.com/mintlayer/mintlayer-core/blob/eedfdae4eb196966908b072e97d39870b8317cb1/.github/scripts/codecheck.py#L141).

I see now, it ignores `todo!` with an argument

Reply:
Maybe we could use clippy to catch this, will be more reliable than grepping code: https://rust-lang.github.io/rust-clippy/stable/index.html#todo

Reply:
If clippy can do it, then it's better.